### ISSUE TITLE
Missing expectations in AbstractLanguageServerTest

### ISSUE THREAD

**Issue by dunnry** [eclipse/xtext-core#1381]
Currently, if you have `Resource `operations in your `WorkspaceEdit`, you will get errors because those operations (`CreateFile`, `CreateFileOptions`, ...) are not implemented.

The current workaround is that you have to override the base class and provide implementations.  E.g. (in Kotlin):

```
override fun toExpectation(it: Any?): String {
	if (it is CreateFile) {
		return _toExpectation(it)
	} else if (it is CreateFileOptions) {
		return _toExpectation(it)
	}
	return super.toExpectation(it)
}

override fun _toExpectation(it: WorkspaceEdit?): String {
	if (it?.changes == null) {
		//this initializes 'changes' correctly
		val edit = WorkspaceEdit()
		//copy over the changes and call base
		edit.documentChanges = it?.documentChanges
		return super._toExpectation(edit)
	}
	return super._toExpectation(it)
}

fun _toExpectation(it: CreateFile?): String {
	return """
		CreateFile: ${super.relativize(it?.uri)}
		Options: ${_toExpectation(it?.options)}
	""".trimIndentLikeXtend()
}

fun _toExpectation(it: CreateFileOptions?): String {
	return """overwrite: ${it?.overwrite}
		ignoreIfExists: ${it?.ignoreIfExists}
	""".trimIndentLikeXtend()
}
```

Note, this also works around issue #1380

**Comment by cdietrich** [comment]
hi, would you come up with a fix?

**Comment by cdietrich** [comment]
Fixed in 2.21. Thx @dunnry
